# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1323390338 18000
#      Thu Dec 08 19:25:38 2011 -0500
# Node ID 0b94080d2b0fc85b8af08f3cc8aa7e8143588d72
# Parent  8feb25ecddaf41765a006f22ec0a33b83cd18363
__go_draw_axes__.m: Fix bug introduced by f35b593688a5 which prevented
multiple "_" and "^" from being rendered literally when "interpreter"
was set to "none".

diff --git a/scripts/plot/private/__go_draw_axes__.m b/scripts/plot/private/__go_draw_axes__.m
--- a/scripts/plot/private/__go_draw_axes__.m
+++ b/scripts/plot/private/__go_draw_axes__.m
@@ -2318,18 +2318,18 @@ function str = no_super_sub_scripts (str
   for marker = "_^" 
     for m = 1 : numel(labels)
       n1 = strfind (labels{m}, sprintf ("\\%s", marker));
       n2 = strfind (labels{m}, marker);
       if (! isempty (n1))
         n1 = n1 + 1;
         n2 = setdiff (n2, n1);
       end
-      for n = n2
-        labels{m} = [labels{m}(1:n2-1), "\\", labels{m}(n2:end)];
+      for n = numel(n2):-1:1
+        labels{m} = [labels{m}(1:n2(n)-1), "\\", labels{m}(n2(n):end)]
       endfor
     endfor
   endfor
   if (iscellstr (str))
     str = labels;
   else
     str = char (labels);
   endif
